pmlc/conversion/pxa_to_affine/pxa_to_affine.cc

```diff
@@ -231,6 +231,10 @@ struct FuncOpConversion : public OpConversionPattern<FuncOp> {
   matchAndRewrite(FuncOp op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const final {
     FunctionType type = op.getType();
+    /* do not rewrite xsmm dispatch functions */
+    if (op.getName().str() == "plaidml_rt_xsmm_dispatch_gemm_f32") {
+      return mlir::success();
+    }
```

Comment:
Might be better/more general to have this rewrite only work on FuncOps with a body .

Reply:
I believe the whole pass right now is a bit weird, why making a functions output part of the inputs?

Reply:
What we are doing here is generally named 'buffer assignment'. The idea is to convert from a functional/value-like semantic into one with buffers and side-effects where we write results into output buffers provided by the user. However, this only needs to take place on the top-level function, not any declarations for external functions provided at runtime (like `plaidml_rt_xsmm_dispatch_gemm_f32`).

@jbruestle WDYT?

Reply:
I think we'll also need to adjust the 
```
target.addDynamicallyLegalOp<FuncOp>(
      [](FuncOp op) { return op.getType().getNumResults() == 0; });
```

So that it also allows for empty function body declarations to be considered legal regardless the number of results.